sms_sponsorship/forms/sms_sponsorship_registration_form.py

```diff
@@ -220,6 +233,9 @@ def form_after_create_or_update(self, values, extra_values):
             sms_request.complete_step2()
             sponsorship.button_generate_invoices()
 
+        def form_after_create_or_update(self, values, extra_values):
+            self.with_delay().finalize_form(self.main_object.sudo())
```

Comment:
Again, I think it makes sense to move the job function inside the `recurring.contract` object so that you can call here `self.main_object.sudo().with_delay().finalize_form()`